src/lib/__tests__/RoutingManager-test.ts

```diff
@@ -3,23 +3,42 @@ import qs from 'qs';
 import instantsearch from '../main';
 import RoutingManager from '../RoutingManager';
 import historyRouter from '../routers/history';
+import {
+  Router,
+  StateMapping,
+  WidgetFactory,
+  Widget,
+  RouteState,
+} from '../../types';
+
+const runAllMicroTasks: () => Promise<any> = () => new Promise(setImmediate);
+
+type FakeSearchClient = {
+  search: (query: string) => Promise<{ results: object[] }>;
+};
 
-const runAllMicroTasks = () => new Promise(setImmediate);
-
-const createFakeSearchClient = () => ({
+const createFakeSearchClient: () => FakeSearchClient = () => ({
   search: () => Promise.resolve({ results: [{}] }),
 });
 
-const createFakeRouter = (args = {}) => ({
-  onUpdate() {},
-  write() {},
+const createFakeRouter: (args: Partial<Router>) => Router = (args = {}) => ({
```

Comment:
```suggestion
const createFakeRouter = (args = {}: Partial<Router>):  Router => ({
```

I forgot the exact syntax, but something more like that makes more sense to me